packages/squiggle-lang/src/rescript/Distributions/GenericDist.res

```diff
@@ -159,6 +119,89 @@ let toPointSet = (
   }
 }
 
+module Score = {
+  type scoreDistOrScalar = DistributionTypes.DistributionOperation.scoreDistOrScalar
+
+  type pointSet_ScoreDistOrScalar = D(PointSetTypes.pointSetDist) | S(float)
+  let argsMake = (
+    ~esti: scoreDistOrScalar,
+    ~answ: scoreDistOrScalar,
+    ~prior: option<scoreDistOrScalar>,
+  ): result<PointSetDist_Scoring.scoreArgs, error> => {
+    let toPointSetFn = toPointSet(
+      ~xyPointLength=MagicNumbers.Environment.defaultXYPointLength,
+      ~sampleCount=MagicNumbers.Environment.defaultSampleCount,
+      ~xSelection=#ByWeight,
+    )
+    let prior': option<result<pointSet_ScoreDistOrScalar, error>> = switch prior {
+    | None => None
+    | Some(Score_Dist(d)) => toPointSetFn(d, ())->E.R.bind(x => x->D->Ok)->Some
+    | Some(Score_Scalar(s)) => s->S->Ok->Some
+    }
+    let twoDists = (esti': t, answ': t): result<
+      (PointSetTypes.pointSetDist, PointSetTypes.pointSetDist),
+      error,
+    > => E.R.merge(toPointSetFn(esti', ()), toPointSetFn(answ', ()))
+    switch (esti, answ, prior') {
+    | (Score_Dist(esti'), Score_Dist(answ'), None) =>
+      twoDists(esti', answ')->E.R.bind(((esti'', answ'')) =>
+        {estimate: esti'', answer: answ'', prior: None}
```

Comment:
I think I'd prefer just redefininig the variable instead of adding an increasing number of `'`s

Reply:
I shorten and use primes because I think shadowing is at least a little smelly, and becomes very smelly when shadowing changes the type of a name.

My priors have to do with 
1. working on python codebases where people excessively shadow, I find those codebases take a ton of effort to read.
2. I wrote a lot of coq (which doesn't even admit shadowing into the theory). 

That said, I'm willing to trust your gut and implement it your way, I'd just like it to be settled and in the style guide moving forward.